python/kserve/kserve/protocol/rest/v2_datamodels.py

```diff
@@ -229,10 +229,11 @@ class InferenceResponse(BaseModel):
         }
     """
     model_name: str
-    model_version: Optional[str]
+    model_version: Optional[str] = None
     id: str
-    parameters: Optional[Parameters]
+    parameters: Optional[Parameters] = None
     outputs: List[ResponseOutput]
+    model_config = ConfigDict(protected_namespaces='')
```

Comment:
Why do we need the `model_config` ?

Reply:
This is the new syntax for pydantic 2 that's analogous to the nested Config class in pydantic 1.

My guess is that Pydantic2 uses "model_" a protected namespace which we need for `model_version`

Reply:
This is to suppress this warning:

```
E   UserWarning: Field "model_name" has conflict with protected namespace "model_".
E
E   You may be able to resolve this warning by setting `model_config['protected_namespaces'] = ()`.
```

So it really ought to be an empty tuple.